sympy/geometry/ellipse.py

```diff
@@ -643,7 +643,17 @@ def intersection(self, o):
                 return self
             else:
                 ellipse_equation = self.equation(x, y)
-                return list(ordered([Point(i) for i in solve([ellipse_equation, o.equation(x, y)], [x, y])]))
+                result = solve([ellipse_equation, o.equation(x, y)], [x, y])
```

Comment:
For me, the bigger question is "Why is solve returning solutions that are imaginary?" This apparently passes through the solution checking there; with x and y real, it shouldn't be returning non-real values if they can (with precision) be shown to be non-real.